psptool/psptool.py

```diff
@@ -122,3 +123,42 @@ def ls_entries(self, entries=None, verbose=False):
             fields += verbose_fields
 
         print(t.get_string(fields=fields))
+
+    def ls_entries_json(self, entries=None, no_duplicates=False, verbose = False):
+        # list all entries of all directories by default (sorted by their address)
+        if entries is None:
+            entries = sorted(self.blob.unique_entries)
+
+        out = []
+        for index, entry in enumerate(entries):
+            info = []
+            if entry.compressed:
+                info.append('compressed')
+            if entry.signed:
+                info.append('signed(%s)' % entry.get_readable_signed_by())
+                if entry.verify_signature():
+                    info.append('verified')
+            if entry.is_legacy:
+                info.append('legacy Header')
+            if entry.encrypted:
+                info.append('encrypted')
+
+            all_values = {
+                'index': index,
+                'address': hex(entry.get_address()),
+                'size': hex(entry.buffer_size),
+                'section_type': entry.get_readable_type(),
+                'magic': entry.get_readable_magic(),
+                'version': entry.get_readable_version(),
+                'info': info,
+                'md5': entry.md5()[:4].upper()
+            }
+
+            if type(entry) is HeaderEntry:
+                all_values['size_signed'] = hex(entry.size_signed)
```

Comment:
This better be another dict here, I think

Reply:
What do you mean? `all_values` is a dict already, right?

Also, do you prefer the JSON output to have address and size as a hex string, or as actual integers?

```json
[
    {
        "index": 0,
        "address": "0x800",
        "size": "0x0",
        "section_type": "FW_GEC~0x61",
        "magic": "",
        "version": "",
        "info": [],
        "md5": "D41D"
    },
```

Reply:
Right, I think plain integers make more sense to avoid extra parsing. :+1: 

What I meant by another dict is nesting, so something like
```json
[
  {
    "index": 9,
    "address": 755456,
    "size": 42368,
    "section_type": "PSP_FW_RECOVERY_BOOT_LOADER~0x3",
    "magic": "$PS1",
    "version": "FF.9.0.6A",
    "info": [
      "signed(1BB9)",
      "verified"
    ],
    "md5": "FE4D",
    "sizes": {
      "signed": 41856,
      "uncompressed": 0,
      "packed": 42368
    }
]
```

Reply:
Oh I see. Feel free to add this, otherwise I will merge it as is.

Reply:
done that 🎉